src/main.rs

```diff
@@ -957,4 +984,102 @@ mod tests {
             Ok(())
         });
     }
+
+    #[test]
+    fn test_virtio_console() {
+        test_block!(tb, "", {
+            let guest = Guest::new();
+            let mut child = Command::new("target/debug/cloud-hypervisor")
+                .args(&["--cpus", "1"])
+                .args(&["--memory", "size=512M"])
+                .args(&["--kernel", guest.fw_path.as_str()])
+                .args(&["--disk", guest.disks[0].as_str(), guest.disks[1].as_str()])
+                .args(&["--net", guest.default_net_string().as_str()])
+                .args(&["--console", "tty"])
+                .stdout(Stdio::piped())
+                .stderr(Stdio::piped())
+                .spawn()
+                .unwrap();
+
+            thread::sleep(std::time::Duration::new(20, 0));
+
+            // We are checking if console device's device id and vendor id pair matches
+            let devices = guest.get_pci_device_ids();
+            let devices: Vec<&str> = devices.split('\n').collect();
+            let vendors = guest.get_pci_vendor_ids();
+            let vendors: Vec<&str> = vendors.split('\n').collect();
+
+            let mut pair_matched = false;
+            for (index, id) in devices.iter().enumerate() {
+                if *id == "0x1043" && Some(&"0x1af4") == vendors.get(index) {
+                    pair_matched = true;
+                    break;
+                }
+            }
+
```

Comment:
Suggestion for a follow-up PR: This whole block could be factorized into a routine that takes a VID and DID as arguments and run for each virtio-pci device we enable

Reply:
I factorized it into a routine according to your suggestion.